### PR TITLE
Fixed issue in the search for points inside the Wigner-Seitz cell

### PR DESCRIPTION
**Pull request by gibertini**
This commit should fix this issue:
https://github.com/wannier-developers/wannier90/issues/210#issue-376332272

When unit cells are very anisotropic (e.g. very stretched in one direction) the original range of points among which the code searches for points inside the Wigner-Seitz cell was insufficient. 

The range is now larger to comply with these cases and it's no longer hard coded, so that in case of need it can be further enlarged by changing the variable search_size in hamiltonian_wigner_seitz.

### CODE REVIEW

File: src/hamiltonian.F90 [hunk 1]
```diff
@@ -578,18 +583,21 @@ subroutine hamiltonian_wigner_seitz(count_pts)
...
+      tot = tot + 1.0_dp/real(ndegen(i), dp)
+    enddo
+
     if (iprint >= 3 .and. on_root) then
       write (stdout, '(1x,i4,a,/)') nrpts, ' lattice points in Wigner-Seitz supercell:'
       do i = 1, nrpts
         write (stdout, '(4x,a,3(i3,1x),a,i2)') '  vector ', irvec(1, i), irvec(2, i), &
           irvec(3, i), '  degeneracy: ', ndegen(i)
       enddo
+      write (stdout, *) ' tot = ', tot
```
Comment (giovannipizzi) [inline src/hamiltonian.F90]: Add an explicit format as in the lines above

File: src/hamiltonian.F90 [hunk 2]
```diff
@@ -505,18 +505,19 @@ subroutine hamiltonian_wigner_seitz(count_pts)
 
     logical, intent(in) :: count_pts
     !! Only count points and return
-
+    integer, parameter :: search_size = 2
```
Comment (giovannipizzi) [inline src/hamiltonian.F90]: As we discussed, make search_size an input parameter, and make a quick benchmark to see, with a dense k-mesh, if 2 is much slower (then leave 1 as default) or 2 is ok.

**Comment by gibertini** [comment]
Following the suggestions by @giovannipizzi I introduced a new input variable (`ws_search_size`). 
This variable is now used consistently in all routines that involve the search for points inside the Wigner-Seitz cell (that is inside `hamiltonian.F90` and `ws_distance.F90`). I have done something similar also for `ws_distance_tol` that is now used also in `hamiltonian.F90`. 
The default value for `ws_search_size` has been set to 2. This does not introduce a serious slow-down of the code, and is safer when using `use_ws_distance = .true.` as the WF centres can wander around in neighbouring unit cells (the previous default in that part of the code was 3). Maybe it would be good to have a feedback from @paulatz on this choice.

**Comment by giovannipizzi** [comment]
Indeed, two questions we need to answer (after discussion with @gibertini today) are:
- is it ok to use the same variable for both? (and is the value a good compromise)?
- Same question after #39 is merged? (even if people might not be using translate_home_cell, and probably we will implement it only for plotting and not for the actual matrices? mentioning also @VVitale for feedback)

If we are unsure, possibly the best approach is to put two different variables, setting the defaults to 2 and 3, respectively. We just need to find good names.

**Comment by gibertini** [comment]
As a compromise, the local value of ws_search_size is incremented by one inside ws_distance.F90.
I added a comment regarding this, both in the source code and in the documentation.

**Pull request merged**